.github/workflows/main_ci.yml

```diff
@@ -45,30 +39,27 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
-    - name: Dependencies (Ubuntu)
-      if: ${{ matrix.os == 'ubuntu-latest' }}
-      run: |
-        sudo apt-get install -y linux-headers-$(uname -r)
+    - uses: lukka/get-cmake@latest
 
-    - name: Restore cache
-      uses: actions/cache@v3
+    - name: Setup vcpkg
+      uses: lukka/run-vcpkg@v10
       with:
-        path: ${{ github.workspace }}/build/cache
-        key: VCPKG-BinaryCache-${{ runner.os }}
+        vcpkgGitCommitId:  ${{ env.VCPKG_COMMIT_ID }}
+        vcpkgDirectory: vcpkg
 
     - name: Build (OpenSSL 1.1)
       run: |
-        cmake -B "${{ env.CMAKE_BUILD_DIR }}" -DTESTING=ON -DCMAKE_TOOLCHAIN_FILE="$VCPKG_INSTALLATION_ROOT/scripts/buildsystems/vcpkg.cmake"
-        cmake --build "${{ env.CMAKE_BUILD_DIR }}" --target all --parallel 2
+        cmake -B build -DTESTING=ON
+        cmake --build build --target all
```

Comment:
I would add the `--parallel 2` back.  Even though we may only have once vcpu, we want to keep the compiler as busy as possible. I have always seen recommendations of cpu.count + 1.